### PR TITLE
Terrain layer API audit

### PR DESCRIPTION
**Pull request by Pessimistress**
For #4317 #4346

#### Change List
- Rename `terrainImage` to `elevationData`
- Rename `surfaceImage` to `texture`
- Docs

### CODE REVIEW

File: docs/layers/terrain-layer.md [hunk 1]
```diff
@@ -140,25 +136,25 @@ The default value of `elevationDecoder` decodes a grayscale image:
 
 ##### `bounds` (Array, optional)
 
-Bounds of the image to fit x,y coordinates into. In [minX, minY, maxX, maxY] world coordinates. Must be supplied when using as a Non-Tiled terrain, otherwise optional.
+Bounds of the image to fit x,y coordinates into. In `[left, top, right, bottom]` world coordinates. Must be supplied when using non-tiled elevation data.
```
Comment (chrisgervang) [inline docs/layers/terrain-layer.md]: The left, top, right, bottom notation is confusing to me. For example, I don’t remember what top means - I think indicating it’s the “min y” is helpful. This is especially confusing when different tools order lat/lng differently.

Comment (Pessimistress) [reply docs/layers/terrain-layer.md]: minY/maxY refer to different things in different context. In a geospatial viewport, the minLat is actually at the bottom.

Comment (chrisgervang) [reply docs/layers/terrain-layer.md]: That makes sense. Would you be able to clarify the relation between min/max lat/lng and top/bottom/left/right in the docs?

Comment (Pessimistress) [reply docs/layers/terrain-layer.md]: Next line in doc

**Pull request merged**